client/my-sites/backup/backup-date-picker/test/index.jsx

```diff
@@ -56,205 +60,161 @@ describe( 'BackupDatePicker', () => {
-			button.simulate( 'click' );
-		} ) );
-
-	test( 'Navigates forward when the next date is clicked', () =>
-		new Promise( ( done ) => {
-			const selectedDate = moment( '2020-01-01' );
-			const onDateChange = ( date ) => {
-				expect( date.diff( selectedDate, 'days' ) ).toEqual( 1 );
-				done();
-			};
-
-			const picker = shallow(
-				<BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } />
-			);
-
-			const button = picker.find( '.backup-date-picker__select-date--next' );
-			button.simulate( 'click' );
-		} ) );
-
-	test( 'Navigates backward when the previous date is focused and the spacebar is pressed', () =>
-		new Promise( ( done ) => {
-			const selectedDate = moment( '2020-01-01' );
-			const onDateChange = ( date ) => {
-				expect( date.diff( selectedDate, 'days' ) ).toEqual( -1 );
-				done();
-			};
-
-			const picker = shallow(
-				<BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } />
-			);
-
-			const button = picker.find( '.backup-date-picker__select-date--previous' );
-			button.simulate( 'keydown', { key: ' ' } );
-		} ) );
-
-	test( 'Navigates forward when the next date is focused and the spacebar is pressed', () =>
-		new Promise( ( done ) => {
-			const selectedDate = moment( '2020-01-01' );
-			const onDateChange = ( date ) => {
-				expect( date.diff( selectedDate, 'days' ) ).toEqual( 1 );
-				done();
-			};
-
-			const picker = shallow(
-				<BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } />
-			);
-
-			const button = picker.find( '.backup-date-picker__select-date--next' );
-			button.simulate( 'keydown', { key: ' ' } );
-		} ) );
+	test( 'Navigates backward when the previous date is clicked', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Dec 31, 2019' );
+		await user.click( button );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.subtract( 1, 'day' ) );
+	} );
+
+	test( 'Navigates forward when the next date is clicked', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Jan 2, 2020' );
+		await user.click( button );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.add( 1, 'day' ) );
+	} );
+
+	test( 'Navigates backward when the previous date is focused and the spacebar is pressed', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Dec 31, 2019' );
```

Comment:
This will actually select the wrapping `span` and not the button. So when we focus and trigger a keydown event on it, this won't do anything. Instead, to focus the button, we could do something like:

```suggestion
		const button = screen.getAllByRole( 'button' )[ 0 ];
```